homeassistant/components/ridwell/coordinator.py

```diff
@@ -0,0 +1,80 @@
+"""Define a Ridwell coordinator."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+from typing import cast
+
+from aioridwell.client import async_get_client
+from aioridwell.errors import InvalidCredentialsError, RidwellError
+from aioridwell.model import RidwellAccount, RidwellPickupEvent
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import LOGGER
+
+UPDATE_INTERVAL = timedelta(hours=1)
+
+
+class RidwellDataUpdateCoordinator(
+    DataUpdateCoordinator[dict[str, RidwellPickupEvent]]
+):
+    """Class to manage fetching data from single endpoint."""
+
+    def __init__(self, hass: HomeAssistant, *, entry: ConfigEntry) -> None:
+        """Initialize global data updater."""
+        self._entry = entry
```

Comment:
The `DataUpdateCoordinator` already has a `config_entry` member, maybe use that one instead?